src/libraries/System.Security.Cryptography.Cose/src/System/Security/Cryptography/Cose/CoseHeaderMap.cs

```diff
@@ -47,59 +75,152 @@ private CoseHeaderMap(bool isReadOnly)
+        /// -or
+        /// The encoded bytes contain trailing data or more than one CBOR value.</exception>
         public int GetValueAsInt32(CoseHeaderLabel label) => _headerParameters[label].GetValueAsInt32();
 
+        /// <summary>
+        /// Gets the CBOR encoded value associated with the specified label, as a text string.
+        /// </summary>
+        /// <param name="label">The label of the value to get.</param>
+        /// <returns>The decoded value associated with the specified label, as a text string.</returns>
+        /// <exception cref="InvalidOperationException">Decodification failed, see the inner exception for the exact exception that was thrown.
+        /// -or
+        /// The encoded bytes contain trailing data or more than one CBOR value.</exception>
         public string GetValueAsString(CoseHeaderLabel label) => _headerParameters[label].GetValueAsString();
 
+        /// <summary>
+        /// Gets the CBOR encoded value associated with the specified label, as a byte string.
+        /// </summary>
+        /// <param name="label">The label of the value to get.</param>
+        /// <returns>The decoded value associated with the specified label, as a byte string.</returns>
+        /// <exception cref="InvalidOperationException">Decodification failed, see the inner exception for the exact exception that was thrown.
+        /// -or
+        /// The encoded bytes contain trailing data or more than one CBOR value.</exception>
         public byte[] GetValueAsBytes(CoseHeaderLabel label) => _headerParameters[label].GetValueAsBytes();
 
+        /// <summary>
+        /// Gets the CBOR encoded value associated with the specified label, as a byte string.
+        /// </summary>
+        /// <param name="label">The label of the value to get.</param>
+        /// <param name="destination">The buffer in which to write the decoded value.</param>
+        /// <returns>The number of bytes written to <paramref name="destination" />.</returns>
+        /// <exception cref="ArgumentException"><paramref name="destination"/> is too small to hold the value.</exception>
+        /// <exception cref="InvalidOperationException">Decodification failed, see the inner exception for the exact exception that was thrown.
+        /// -or
+        /// The encoded bytes contain trailing data or more than one CBOR value.</exception>
         public int GetValueAsBytes(CoseHeaderLabel label, Span<byte> destination) => _headerParameters[label].GetValueAsBytes(destination);
 
+        /// <summary>
+        /// Adds the specified key and value to the dictionary.
+        /// </summary>
+        /// <param name="key">The key of the element to add.</param>
+        /// <param name="value">The value of the element to add.</param>
+        /// <exception cref="InvalidOperationException">The <see cref="CoseHeaderMap"/> is read-only.</exception>
```

Comment:
Either the English

```suggestion
        /// <exception cref="InvalidOperationException">The header map is read-only.</exception>
```

or the robotic (but useful)

```suggestion
        /// <exception cref="InvalidOperationException"><see cref="IsReadOnly"/> is <see langword="true"/>.</exception>
```